File: projecteuler/python/problem_1.py
'''
Created on 22/08/2011

@author: Breno Martinusso <breno@martinusso.com>

Problem 1
05 October 2001

If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
'''

s = 0
for i in xrange(1, 1000):# In Python 3 use range()
    s+= i if not i % 3 or not i % 5 else 0
print s

File: projecteuler/go/problem1.go
package projecteuler

// Problem 1
// Multiples of 3 and 5
// If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
// Find the sum of all the multiples of 3 or 5 below 1000.

const (
	answerProblem1 = 233168
)

func problem1() int64 {
	sum := 0
	for n := 3; n < 1000; n++ {
		if isMultiplesOf3And5(n) {
			sum += n
		}
	}
	return int64(sum)
}

func isMultiplesOf3And5(number int) bool {
	return (number%3 == 0) || (number%5 == 0)
}
